sonar-python-plugin/src/main/java/org/sonar/plugins/python/ruff/RuffJsonReportReader.java

```diff
@@ -81,22 +81,22 @@ private void onResult(JSONObject result) {
     issue.startLocationCol = toInteger(location.get("column"));
     issue.startLocationRow = toInteger(location.get("row"));
     JSONObject endLocation = (JSONObject) result.get("end_location");
-    issue.endLocationCol = correctEndLocationCol(endLocation.get("column"), issue.startLocationCol);
     issue.endLocationRow = toInteger(endLocation.get("row"));
+    issue.endLocationCol = correctEndLocationCol(endLocation.get("column"), issue.startLocationCol, issue.startLocationRow, issue.endLocationRow);
     consumer.accept(issue);
   }
 
   /*
     Ruff returns the col number of the last char + 1.
     In order to properly read the col number we need to return the number of the last char.
    */
-  private static Integer correctEndLocationCol(Object value, int startLocationCol) {
+  private static Integer correctEndLocationCol(Object value, int startLocationCol, int startLocationRow, int endLocationRow) {
```

Comment:
I assume that in this method we already assume that `startLocationRow <= endLocationRow`? Just making sure, as I'm not entirely clear on this.

Reply:
Yes so on the same row we can get the end location column which is either bigger or equal to the start location. For example, on an empty line, we get the same end location and start location